src/term.rs

```diff
@@ -157,19 +177,48 @@ impl Term {
         })
     }
 
+    /// Return a terminal for the given Read/Write pair styled-like Stderr.
+    #[cfg(unix)]
+    pub fn read_write_pair<R, W>(read: R, write: W) -> Term
+    where
+        R: io::Read + std::fmt::Debug + AsRawFd + Send + 'static,
+        W: Write + std::fmt::Debug + AsRawFd + Send + 'static,
+    {
+        Self::read_write_pair_with_style(read, write, Style::new().for_stderr())
+    }
+
+    /// Return a terminal for the given Read/Write pair.
+    #[cfg(unix)]
+    pub fn read_write_pair_with_style<R, W>(read: R, write: W, style: Style) -> Term
+    where
+        R: io::Read + std::fmt::Debug + AsRawFd + Send + 'static,
+        W: Write + std::fmt::Debug + AsRawFd + Send + 'static,
+    {
+        Term::with_inner(TermInner {
+            target: TermTarget::ReadWritePair(ReadWritePair {
+                read: Arc::new(Mutex::new(read)),
+                write: Arc::new(Mutex::new(write)),
+                style,
+            }),
+            buffer: None,
+        })
+    }
+
     /// Returns the style for the term
     #[inline]
     pub fn style(&self) -> Style {
-        match self.target() {
+        match self.inner.target {
             TermTarget::Stderr => Style::new().for_stderr(),
             TermTarget::Stdout => Style::new().for_stdout(),
+            #[cfg(unix)]
+            TermTarget::ReadWritePair(ReadWritePair { ref style, .. }) => style.clone(),
         }
     }
 
     /// Returns the target
     #[inline]
-    pub fn target(&self) -> TermTarget {
-        self.inner.target
+    pub fn target(&self) -> &TermTarget {
+        &self.inner.target
```

Comment:
I dont think we can clone here, right?

Reply:
We can clone here... I'll swap it back to preserve backwards compatibility.

In general though: I think that it would be more idiomatic moving forward to use references in more cases, and then letting the caller clone if they need to.